File: flutter_app/helpdesk_app/lib/models/ticket_comment.dart
class TicketComment {
  const TicketComment({
    required this.id,
    required this.ticketId,
    required this.authorId,
    required this.authorRole,
    required this.body,
    this.isInternal = false,
    required this.createdAt,
  });

  final int id;
  final int ticketId;
  final int authorId;
  final String authorRole;
  final String body;
  final bool isInternal;
  final DateTime createdAt;

  factory TicketComment.fromJson(Map<String, dynamic> json) {
    return TicketComment(
      id: json['id'] as int,
      ticketId: json['ticket_id'] as int,
      authorId: json['author_id'] as int,
      authorRole: (json['author_role'] as String?) ?? '',
      body: (json['body'] as String?) ?? '',
      isInternal: (json['is_internal'] as bool?) ?? false,
      createdAt:
          DateTime.tryParse((json['created_at'] as String?) ?? '') ??
          DateTime.now(),
    );
  }
}

File: backend/ticket-service/internal/domain/ticket_comment.go
package domain

import "time"

// TicketComment is one message in a ticket's conversation thread.
// IsInternal marks a staff-only note — never visible to the ticket's
// owning user, whether via GET /comments or a push notification.
type TicketComment struct {
	ID         uint      `gorm:"primaryKey" json:"id"`
	TicketID   uint      `json:"ticket_id"`
	AuthorID   uint      `json:"author_id"`
	AuthorRole string    `json:"author_role"`
	Body       string    `json:"body"`
	IsInternal bool      `gorm:"default:false" json:"is_internal"`
	CreatedAt  time.Time `json:"created_at"`
}

type CommentRepository interface {
	Create(comment *TicketComment) error
	FindByTicket(ticketID uint) ([]TicketComment, error)
}
